cmd/jiralert/main.go

```diff
@@ -38,64 +43,84 @@ func main() {
 		runtime.SetMutexProfileFraction(1)
 	}
 
-	// Override -alsologtostderr default value.
-	if alsoLogToStderr := flag.Lookup("alsologtostderr"); alsoLogToStderr != nil {
-		alsoLogToStderr.DefValue = "true"
-		alsoLogToStderr.Value.Set("true")
-	}
 	flag.Parse()
 
-	log.Infof("Starting JIRAlert version %s", Version)
+	var logger log.Logger
+	{
+		var lvl level.Option
+		switch *logLevel {
+		case "error":
+			lvl = level.AllowError()
+		case "warn":
+			lvl = level.AllowWarn()
+		case "debug":
+			lvl = level.AllowDebug()
+		case "info":
+		default:
+			lvl = level.AllowInfo()
+		}
+		logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+		if *logFormat == logFormatJson {
+			logger = log.NewJSONLogger(log.NewSyncWriter(os.Stderr))
+		}
+		logger = level.NewFilter(logger, lvl)
+
+		logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
+	}
 
-	config, _, err := config.LoadFile(*configFile)
+	level.Info(logger).Log("msg", "starting JIRAlert", "version", Version)
```

Comment:
well yes and no (: but the fact it used in *most* of Prometheus related code, makes it the default choice. It is bit tricky, epecially that it:
* fails if have odd number of args
* fails if something is not string (or has no stringer)

But otherwise it's a casual structured logger, better then glog definintely (IMO). The is also rule that those log fields should be lower case and not full sentences (no fullstop). No sure where that rule was written though ;p

Reply:
I was considering using capitalized string within `msg`, but looking into [the Prometheus code](https://github.com/prometheus/prometheus/blob/master/storage/remote/wal_watcher.go#L142), it seems to be lowercase. Since most other parameters are going to be lowercase, I think it make sense to keep `msg` lowercase as well.